src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UICaseSearchController.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.ccd.v2.internal.controller;
+
+import io.swagger.annotations.*;
+import lombok.extern.slf4j.*;
+import org.springframework.beans.factory.annotation.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.hmcts.ccd.auditlog.AuditOperationType;
+import uk.gov.hmcts.ccd.auditlog.LogAudit;
+import uk.gov.hmcts.ccd.domain.model.search.*;
+import uk.gov.hmcts.ccd.domain.model.search.elasticsearch.SearchResultViewItem;
+import uk.gov.hmcts.ccd.domain.model.search.elasticsearch.UICaseSearchResult;
+import uk.gov.hmcts.ccd.domain.service.search.elasticsearch.*;
+import uk.gov.hmcts.ccd.domain.service.search.elasticsearch.security.*;
+import uk.gov.hmcts.ccd.v2.*;
+import uk.gov.hmcts.ccd.v2.internal.resource.*;
+
+import java.time.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.auditlog.aop.AuditContext.CASE_ID_SEPARATOR;
+import static uk.gov.hmcts.ccd.auditlog.aop.AuditContext.MAX_CASE_IDS_LIST;
+
+@RestController
+@RequestMapping(path = "/internal/searchCases")
+@Slf4j
+public class UICaseSearchController {
```

Comment:
minor: 
we could call it `UISearchController` so that we can add other types of search related endpoints in the future, for example if we introduced searching events.